src/api/app/helpers/status/required_checks_helper.rb

```diff
@@ -4,6 +4,8 @@ def build_header(project, checkable)
       { project: project.name, repository: checkable.name }
     elsif checkable.is_a?(Package)
       { project: project.name, package: checkable.name }
+    elsif checkable.is_a?(RepositoryArchitecture)
```

Comment:
would you mind to try it with when/case?

Reply:
clearly out of scope

Reply:
I don't see how can it be out of the scope if you are changing it...  just copy and paste it if you want:

```
def build_header(project, checkable)
   case checkable
   when Repository
      { project: project.name, repository: checkable.name }
   when Package
      { project: project.name, package: checkable.name }
   when RepositoryArchitecture
      build_header(project, checkable.repository).merge(architecture: checkable.architecture.name)
   else
      { project: checkable.name }
    end
  end
```

Reply:
feel free to push into the branch